src/game/server/ddracecommands.cpp

```diff
@@ -827,3 +827,50 @@ void CGameContext::ConDumpAntibot(IConsole::IResult *pResult, void *pUserData)
 	CGameContext *pSelf = (CGameContext *)pUserData;
 	pSelf->Antibot()->Dump();
 }
+
+void CGameContext::ConDumpLog(IConsole::IResult *pResult, void *pUserData)
+{
+	CGameContext *pSelf = (CGameContext *)pUserData;
+	int LimitSecs = MAX_LOG_SECONDS;
+	if(pResult->NumArguments() > 0)
+		LimitSecs = pResult->GetInteger(0);
+
+	if(LimitSecs < 0)
+		return;
+
+	for(auto &m_aLog : pSelf->m_aLogs)
+	{
+		if(!m_aLog.m_Timestamp)
+			continue;
+
+		int Seconds = (time_get() - m_aLog.m_Timestamp) / time_freq();
+		if(Seconds > LimitSecs)
+			continue;
+
+		char aBuf[256];
+		if(m_aLog.m_Server)
+			str_format(aBuf, sizeof aBuf, "%s, %i seconds ago", m_aLog.m_aDescription, Seconds);
+		else
+			str_format(aBuf, sizeof aBuf, "%s, %i seconds ago < addr=<{%s}> name='%s' client=%i",
+				m_aLog.m_aDescription, Seconds, m_aLog.m_aAddrStr, m_aLog.m_aName, m_aLog.m_Client);
+		pSelf->Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "log", aBuf);
+	}
+}
+
+void CGameContext::LogEvent(const char *Description, int ClientID)
+{
+	if(ClientID < 0)
+		m_aLogs[m_LastLog].m_Server = true;
+	else
+	{
+		char aAddrStr[NETADDR_MAXSTRSIZE];
+		m_aLogs[m_LastLog].m_Server = false;
+		m_aLogs[m_LastLog].m_Client = Server()->GetClientVersion(ClientID);
+		Server()->GetClientAddr(ClientID, aAddrStr, NETADDR_MAXSTRSIZE);
+		str_copy(m_aLogs[m_LastLog].m_aAddrStr, aAddrStr, sizeof m_aLogs[m_LastLog].m_aAddrStr);
```

Comment:
```suggestion
		Server()->GetClientAddr(ClientID, m_aLogs[m_LastLog].m_aAddrStr, sizeof(m_aLogs[m_LastLog].m_aAddrStr));
```
No reason for the extra copy. Pleace load the address directly into the right place.